samples/net/zperf/src/zperf_tcp_receiver.c

```diff
@@ -143,21 +145,36 @@ void zperf_tcp_receiver_init(const struct shell *shell, int port)
 			return;
 		}
 
-		ret = zperf_get_ipv4_addr(shell, MY_IP4ADDR,
-					  &in4_addr_my->sin_addr);
-		if (ret < 0) {
-			shell_fprintf(shell, SHELL_WARNING,
-				      "Unable to set IPv4\n");
-			return;
+		if (MY_IP4ADDR) {
```

Comment:
Why did you removed the empty string check?
Shouldn't this check look like
```
if (MY_IP4ADDR && MY_IPV4ADDR[0])
```

Reply:
zephyr/samples/net/zperf/src/zperf_udp_receiver.c:329:30: warning: pointer of type ‘void *’ used in arithmetic [-Wpointer-arith]
if (MY_IP4ADDR && MY_IP4ADDR[0]) {
^
zephyr/samples/net/zperf/src/zperf_udp_receiver.c:329:30: error: void value not ignored as it ought to be
if (MY_IP4ADDR && MY_IP4ADDR[0]) {

Reply:
Hi @jukkar , could you pls give any idea?

Reply:
As you said the MY_IP4ADDR can be set to NULL so the check cannot be done as I suggested. So we can do it like you proposed.